### PR TITLE
Avoid temporary String object creation in StreamMessageProducer

### PR DESCRIPTION
**Pull request by sebthom**
This can reduce the likeliness of OOMs when dealing with large response messages from LS servers. See discussion at https://github.com/eclipse-lsp4j/lsp4j/issues/815

### CODE REVIEW

**Review by jonahgraham — APPROVED** [review]
LGTM - I have trouble imagining it will fully solving your OOM, but saving the 160M is definitely worthwhile.

I'll leave it open for now for others to have a look. But assuming no dissenting views it can be merged soonish.

**Review by pisv — APPROVED** [review]

**Comment by jonahgraham** [comment]
> I'll leave it open for now for others to have a look. But assuming no dissenting views it can be merged soonish.

Thanks to @pisv we have a positive review - even better than simply no dissenting views! Waiting to hear from @sebthom if this commit or #817 should be submitted first (as I assume some amount of merging conflict resolution may be required)

**Comment by sebthom** [comment]
> > I'll leave it open for now for others to have a look. But assuming no dissenting views it can be merged soonish.
> 
> Thanks to @pisv we have a positive review - even better than simply no dissenting views! Waiting to hear from @sebthom if this commit or #817 should be submitted first (as I assume some amount of merging conflict resolution may be required)

Rebasing this PR is easier, so I would prefer if #817 gets merged first.

**Comment by jonahgraham** [comment]
Thanks @sebthom - I'll hold off until @pisv reviews #817 

@pisv feel free to submit #817 if you approve (and #816 once it is rebased)

**Pull request merged**
